compiler-core/src/erlang.rs

```diff
@@ -1375,14 +1375,32 @@ fn bare_clause_guard<'a>(guard: &'a TypedClauseGuard, env: &mut Env<'a>) -> Docu
 
         ClauseGuard::FieldAccess {
             container, index, ..
-        } => tuple_index_inline(container, index.expect("Unable to find index") + 1, env),
+        } => handle_guard_field_access(container, index.expect("Unable to find index"), env),
 
         ClauseGuard::ModuleSelect { literal, .. } => const_inline(literal, env),
 
         ClauseGuard::Constant(constant) => const_inline(constant, env),
     }
 }
 
+fn handle_guard_field_access<'a>(
+    container: &'a TypedClauseGuard,
+    index: u64,
+    env: &mut Env<'a>,
+) -> Document<'a> {
+    match container {
```

Comment:
Maybe add a comment here, explaining what you're doing? It might not be clear why this function exists

Reply:
I may have gone overboard with the comment. Lemme know if I should tone it back a bit lol

Reply:
Probably the first paragraph is fine on its own